app/views/equipment/show.html.erb

```diff
@@ -1,16 +1,32 @@
 <%
-  # TODO: refactor this
+  def reservation_title(res)
+    if user_signed_in? && current_user.id == res.user_id
+      title = "#{current_user.given_name} #{current_user.last_name}"
+      "#{title} (pendiente)" if res.status == 'pending'
+    else
+      'Reservación'
+    end
+  end
+
   available_hours = @equipment.available_hours
 
   reservations = []
   res = @equipment.reservations
-  res = res.confirmed.or(res.blocked).or(res.pending)
+  viable_reservations = res.confirmed.or(res.blocked).or(res.pending)
+
+  res = viable_reservations.select do |reservation|
+    reservation.status == 'pending' ? (reservation.user == current_user) : true
+  end
+
   res.each do | r |
+    res_title = reservation_title(r)
+    res_color = reservation_color(r)
```

Comment:
Something related to this line is breaking the links to enter the equipment view.

Reply:
I was going to replace the method call to reservation_color with the method contents (it was a one liner). I effectively deleted the method, but forgot to replace the method code with the code, so it was crashing. I've since fixed it, thanks for catching it!